consensus/src/quorum_store/utils.rs

```diff
@@ -718,25 +858,37 @@ impl ProofQueue {
         let start = Instant::now();
         for batch in &batches {
             let batch_key = BatchKey::from_info(batch);
-            if let Some(Some((proof, insertion_time))) = self.batch_to_proof.get(&batch_key) {
-                counters::pos_to_commit(
-                    proof.gas_bucket_start(),
-                    insertion_time.elapsed().as_secs_f64(),
-                );
-                self.dec_remaining(&batch.author(), batch.num_txns());
-            }
-            self.batch_to_proof.insert(batch_key.clone(), None);
-            if let Some((txn_summaries, _, true)) = self.batch_summaries.get(&batch_key) {
-                for txn_summary in txn_summaries {
-                    if let Some(count) = self.txn_summary_num_occurrences.get_mut(txn_summary) {
-                        *count -= 1;
-                        if *count == 0 {
-                            self.txn_summary_num_occurrences.remove(txn_summary);
-                        }
-                    };
+            if let Some(item) = self.items.get(&batch_key) {
+                if let Some(ref proof) = item.proof {
+                    let insertion_time = item
+                        .proof_insertion_time
+                        .expect("Insertion time is updated with proof");
+                    counters::pos_to_commit(
+                        proof.gas_bucket_start(),
+                        insertion_time.elapsed().as_secs_f64(),
+                    );
+                    self.dec_remaining(&batch.author(), batch.num_txns());
+                }
+                // TODO(ibalajiarun): This assignment doesn't account for inserts after commits.
+                let item = self
+                    .items
+                    .get_mut(&batch_key)
+                    .expect("must exist due to check");
+                item.proof = None;
+                item.proof_insertion_time = None;
+
+                if let Some(ref txn_summaries) = item.txn_summaries {
```

Comment:
We add a batch's transaction summaries to `txn_summary_num_occurrences` only if the batch has a proof in proof queue. 
If we received the batch's transaction summaries but not the proof, then `txn_summary_num_occurrences` is not counting those transactions. We shouldn't be subtracting those transaction summary counts here.

Reply:
Good catch.